---
title: "NGS Report"
output: html_document
---
```{r, include=FALSE}
library(knitr)
library(plotly)
library(tidyverse)
```
# NGS project report

* This section of the report should contain a few sentences describing the experimental setup.
* This should include information about the biological questions being asked, treatments and controls, replicates, sequencing depth and technology used, etc.

## Raw reads summary table (RNA-seq, DNA-seq, ChIP-seq, ATAC-seq)
* This section of the report should contain a table containing a summary of the initial FastQC run.
```{r, include=FALSE}

multi_qc_directory <- ('/Users/jwalla12/QC_Test_files/raw/multiqc_data/')

picard_directory <- ('/Users/jwalla12/analysis_projects/Scaffa_project/Scaffa_picardtools_metrics/')
```

multiqc_general_stats
######
## General Stats (RNA-seq, DNA-seq, ChIP-seq, ATAC-seq)
```{r, include=FALSE}
### Basic Stats ###
general_stats_func<- function(multi_qc_directory){
  general_stats <- read.table((paste0(multi_qc_directory,'multiqc_general_stats.txt')), header=TRUE, sep='\t')
  general_stats_md <- kable(general_stats, format='markdown')
  return(general_stats_md)
}
```

```{r}
general_stats_func(multi_qc_directory)
```
######

######
## Per base quality plot (RNA-seq, DNA-seq, ChIP-seq, ATAC-seq)
```{r, include=FALSE}
### Per base quality ###
per_base_quality_func <- function(multi_qc_directory){
per_base_quality <- read.table((paste0(multi_qc_directory,'mqc_fastqc_per_base_sequence_quality_plot_1.txt')), header=TRUE, sep='\t')
per_base_quality_tidy <- gather(per_base_quality, position, phred_score, -Sample) 
per_base_quality_tidy$position <- gsub('X','', per_base_quality_tidy$position)
per_base_quality_tidy$position <- sapply(per_base_quality_tidy$position, as.numeric)
per_base_quality_tidy_ggplot <- ggplotly(ggplot(per_base_quality_tidy, aes(x=position, y=phred_score, color=Sample)) + geom_line() + theme_classic())
return(per_base_quality_tidy_ggplot)
}
```

```{r}
per_base_quality_func(multi_qc_directory)
```
######

######
## N-content plot (RNA-seq, DNA-seq, ChIP-seq, ATAC-seq)
```{r, include=FALSE}
### N content ###
per_base_n_content_func <- function(multi_qc_directory){
per_base_n_content <- read.table((paste0(multi_qc_directory,'mqc_fastqc_per_base_n_content_plot_1.txt')), header=TRUE, sep='\t')
per_base_n_content_tidy <- gather(per_base_n_content, position, percent_N, -Sample)
per_base_n_content_tidy$position <- gsub('X','', per_base_n_content_tidy$position)
per_base_n_content_tidy$position <- sapply(per_base_n_content_tidy$position, as.numeric)
per_base_n_content_tidy_ggplot <- ggplotly(ggplot(per_base_n_content_tidy, aes(x=position, y=percent_N, color=Sample)) + geom_line() + theme_classic())
return(per_base_n_content_tidy_ggplot)
}
```

```{r}
per_base_n_content_func(multi_qc_directory)
```
######

######
## GC content plot (RNA-seq, DNA-seq, ChIP-seq, ATAC-seq)
```{r, include=FALSE}
### GC content ###
gc_content_func <- function(multi_qc_directory){
gc_content <- read.table((paste0(multi_qc_directory,'mqc_fastqc_per_sequence_gc_content_plot_Counts.txt')), header=TRUE, sep='\t')

gc_content_tidy <- gather(gc_content, position, count, -Sample)
gc_content_tidy$position <- gsub('X','', gc_content_tidy$position)
gc_content_tidy$position <- sapply(gc_content_tidy$position, as.numeric)
gc_content_tidy_ggplot <- ggplotly(ggplot(gc_content_tidy, aes(x=position, y=count, color=Sample)) + geom_line() + theme_classic())
return(gc_content_tidy_ggplot)
}
```

```{r}
gc_content_func(multi_qc_directory)
```
######

######
## Adapter content plot (RNA-seq, DNA-seq, ChIP-seq, ATAC-seq)
```{r, include=FALSE}
### Adapter content ###
adapter_content_func <- function(multi_qc_directory){
adapter_content <- read.table((paste0(multi_qc_directory,'mqc_fastqc_adapter_content_plot_1.txt')), header=TRUE, sep='\t')

adapter_content_tidy <- gather(adapter_content, position, percent_of_sequences, -Sample)
adapter_content_tidy$position <- gsub('X','', adapter_content_tidy$position)
adapter_content_tidy$position <- sapply(adapter_content_tidy$position, as.numeric)
adapter_content_tidy_ggplot <- ggplotly(ggplot(adapter_content_tidy, aes(x=position, y=percent_of_sequences, color=Sample)) + geom_line() + theme_classic() + theme(legend.position='right'))
return(adapter_content_tidy_ggplot)
}
```

```{r}
adapter_content_func(multi_qc_directory)
```
######


######
## WGS summary statistics (RNA-seq, DNA-seq, ChIP-seq, ATAC-seq)
```{r, include=FALSE}
### WGS summary statistics ###
wgs_summary_statistics_func <- function(picard_directory){
  wgs_metrics_files <- list.files(path = picard_directory, pattern = 'CollectWgsMetrics.txt', full.names=TRUE)
  wgs_metrics <- lapply(wgs_metrics_files, function(x) (read.table(x, skip=6, nrows=1, header=TRUE, fill=TRUE, sep='\t')))
  names(wgs_metrics) <- strsplit(wgs_metrics_files, split='/') %>% sapply(tail, 1) %>% gsub(pattern='.txt', replacement='')
  wgs_metrics_table <- map_df(wgs_metrics, ~as.data.frame(.x))
  wgs_metrics_table$library <- names(wgs_metrics)
  wgs_metrics_md <- kable(wgs_metrics_table, format='markdown')
  return(wgs_metrics_md)
  }
```
```{r}
wgs_summary_statistics_func(picard_directory)
```

  
  
  
  
  
  
  
  
  
  
  InsertSizeMetrics <- lapply(InsertSizeMetrics_files, function(CollectInsertSizeMetrics_file){
  CollectInsertSizeMetrics_table <- read.table(CollectInsertSizeMetrics_file, skip=6, nrows=2, header=TRUE, sep='\t')
  CollectInsertSizeMetrics_table$LIBRARY <- CollectInsertSizeMetrics_file
  CollectInsertSizeMetrics_table_df <- map_df(CollectInsertSizeMetrics_table, ~as.data.frame(.x))
  #separate out the histogram data
  CollectInsertSizeMetrics_table_df_no_histo <- CollectInsertSizeMetrics_table_df[!is.na(CollectInsertSizeMetrics_table_df$MEAN_INSERT_SIZE), ]
  CollectInsertSizeMetrics_table_df_histo <- CollectInsertSizeMetrics_table_df[is.na(CollectInsertSizeMetrics_table_df$MEAN_INSERT_SIZE), ]
  return(CollectInsertSizeMetrics_table_df_no_histo)
}
  
  
  
  
  
  wgs_metrics$LIBRARY <- 
  wgs_table <- bind_rows(wgs_metrics)                      
  insert_size_metrics_rows$LIBRARY<-insert_size_metrics_file
    per_base_sequence_quality <- lapply(entire_qc_files, function(x) (df_rows <- slice(x, (which(x$V1=='>>Per base sequence quality')+4):which(x$V1=='>>Per sequence quality scores')-2) %>% select(c('V1','V2','V3','V4','V5','V6','V7')) %>% rename(Base=V1, Mean=V2, Median=V3, LowerQuartile=V4, UpperQuartile=V5, Percentile_10th=V6, Percentile_90th=V7)))
  
}

```

adapter_content <- lapply(entire_qc_files, function(x) slice(x, (which(x$V1=='>>Adapter Content')+1):(nrow(x)-1))) 


InsertSizeMetrics <- lapply(InsertSizeMetrics_files, function(CollectInsertSizeMetrics_file){
  CollectInsertSizeMetrics_table <- read.table(CollectInsertSizeMetrics_file, skip=6, nrows=2, header=TRUE, sep='\t')
  CollectInsertSizeMetrics_table$LIBRARY <- CollectInsertSizeMetrics_file
  CollectInsertSizeMetrics_table_df <- map_df(CollectInsertSizeMetrics_table, ~as.data.frame(.x))
  #separate out the histogram data
  CollectInsertSizeMetrics_table_df_no_histo <- CollectInsertSizeMetrics_table_df[!is.na(CollectInsertSizeMetrics_table_df$MEAN_INSERT_SIZE), ]
  CollectInsertSizeMetrics_table_df_histo <- CollectInsertSizeMetrics_table_df[is.na(CollectInsertSizeMetrics_table_df$MEAN_INSERT_SIZE), ]
  return(CollectInsertSizeMetrics_table_df_no_histo)
}











<!-- ## Part 5: Kmer content plots -->
<!-- * These are additional outputs from FastQC, images should be stored in the assets folder. -->
<!-- | **Library** | **Raw reads**  | **Trimmed reads** | -->
<!-- |:------:|:-----------:|:----------:| -->
<!-- | Experiment_A_1 | ![](assets/raw_reads/Experiment_A_1/kmer_profiles.png) | ![](assets/trimmed_reads/Experiment_A_1/kmer_profiles.png) | -->

<!-- ## Part 6: Read QC summaryD -->
<!-- * A few sentences here describing the QC results -- do any libraries look particularly unusual? -->

<!-- ## Part 7: Read alignment -->
<!-- * A few sentences here describing how alignments were performed. -->
<!-- * Include which reference was used (where it was downloaded, what version of patch #, etc.) -->
<!-- * Also include the software used for indexing and aligning reads, as well as parameters and flags. -->
<!-- * Include metrics reported by PicardTools as well as the definition of each metric. -->

<!-- ## Part 8: Alignment summary statistics -->
<!-- | **LIBRARY** | **TOTAL_READS** | **READS_ALIGNED_IN_PAIRS** | -->
<!-- |:---:|:---:|:---:| -->
<!-- | Experiment_A_1 | 120000000 |  110000000 | -->

<!-- * LIBRARY:	The library for which statistics are reported. -->
<!-- * TOTAL_READS:	The total number of reads including all PF and non-PF reads. When CATEGORY equals PAIR this value will be 2x the number of clusters. -->
<!-- * READS_ALIGNED_IN_PAIRS:	The number of aligned reads whose mate pair was also aligned to the reference. -->

<!-- ## Part 9: WGS summary statistics -->
<!-- | **LIBRARY** | **MEAN_COVERAGE** | **SD_COVERAGE** | **PCT_EXC_MAPQ** | **PCT_EXC_UNPAIRED** | **PCT_EXC_BASEQ** | **PCT_EXC_OVERLAP** | **PCT_EXC_CAPPED** | **PCT_EXC_TOTAL** | -->
<!-- |:-------:| :-------: | :-------: | :-------: | :-------: | :-------: | :-------: | :-------: | :-------: | :-------: | -->
<!-- | Experiment_A_1 |  3.74 | 3.42 | 0.11  | 1.10E-04 | 7.14E-03 | 0.27 | 0.011 | 0.40 | -->

<!-- * LIBRARY:	The library for which statistics are reported. -->
<!-- * MEAN_COVERAGE:	The mean coverage in bases of the genome territory, after all filters are applied. -->
<!-- * SD_COVERAGE:	The standard deviation of coverage of the genome after all filters are applied. -->
<!-- * MEDIAN_COVERAGE:	The median coverage in bases of the genome territory, after all filters are applied. -->
<!-- * PCT_EXC_MAPQ:	The fraction of aligned bases that were filtered out because they were in reads with low mapping quality (default is < 20). -->
<!-- * PCT_EXC_DUPE:	The fraction of aligned bases that were filtered out because they were in reads marked as duplicates. -->
<!-- * PCT_EXC_UNPAIRED:	The fraction of aligned bases that were filtered out because they were in reads without a mapped mate pair. -->
<!-- * PCT_EXC_BASEQ:	The fraction of aligned bases that were filtered out because they were of low base quality (default is < 20). -->
<!-- * PCT_EXC_OVERLAP:	The fraction of aligned bases that were filtered out because they were the second observation from an insert with overlapping reads. -->
<!-- * PCT_EXC_CAPPED:	The fraction of aligned bases that were filtered out because they would have raised coverage above the capped value (default cap = 250x). -->
<!-- * PCT_EXC_TOTAL:	The total fraction of aligned bases excluded due to all filters. -->

<!-- ## Part 10: Duplication statistics -->
<!-- | **LIBRARY**  | **READ_PAIRS_EXAMINED** | **SECONDARY_OR_SUPPLEMENTARY_RDS** | **UNMAPPED_READS**  | **PERCENT_DUPLICATION** | **ESTIMATED_LIBRARY_SIZE** | -->
<!-- |  :-------:  | :-------: |  :-------:  |  :-------:  |  :-------:  |  :-------:  | -->
<!-- | Experiment_A_1  | 59776401 | 681633 | 659975   | 0.16 | 178349158 | -->

<!-- * LIBRARY:	The library on which the duplicate marking was performed. -->
<!-- * READ_PAIRS_EXAMINED:	The number of mapped read pairs examined. -->
<!-- * SECONDARY_OR_SUPPLEMENTARY_RDS:	The number of reads that were either secondary or supplementary -->
<!-- * UNMAPPED_READS: The total number of unmapped reads examined. -->
<!-- * PERCENT_DUPLICATION:	The fraction of mapped sequence that is marked as duplicate. -->
<!-- * ESTIMATED_LIBRARY_SIZE:	The estimated number of unique molecules in the library based on PE duplication. -->

<!-- ## Part 11: Alignment Summary -->
<!-- * A few sentences here summarizing the alignments -- do any of the libraries look unusual? -->

<!-- ## Part 12: Peak calling using MACS2 -->
<!-- * Describe the parameters used for peak calling, including which libraries were treatment and which were control. -->

<!-- ## Part 13: Overlapping peaks -->
<!-- * Describe the overlapping peaks protocol used -- what command was used, which experiments were compared, etc. -->

<!-- ## Part 14: Annotate HO1-specific peaks -->
<!-- * How were peaks annotated? Which file was used to create a txdb? -->

<!-- ## Part 15: Testing for significance of peaks -->
<!-- * Plot the distibution of q-values across all peaks -->

<!-- ### GroupA q-value distribution -->
<!-- ![](assets/Experiment_A_1-log10qValue_histogram.png) -->

<!-- ## Part 16: Peak locations in the context of genomic features -->
<!-- * To determine the peak locations in the context of genomic features, we used ChipSeeker `plotAnnoPie` and `plotDisttoTSS`. -->

<!-- ### **plotAnnoPie** -->

<!-- ![](assets/Experiment_A_1_peaks_plotAnnoPie.png) -->

<!-- ### **plotDisttoTSS** -->

<!-- ![](assets/Experiment_A_1_peaks_plotDistToTSS.png) -->

<!-- ## Part 17: Distribution and width of peaks across chromosomes -->
<!-- * Plot histograms to get a better understanding of the distribution of peaks across chromosomes. -->

<!-- ![](assets/peaks_per_chromosome_Experiment_A_1_specific.png) -->

<!-- ![](assets/peak_width_per_chromosome_Experiment_A_1_specific.png) -->

<!-- ## Part 18: Summary -->
<!-- * Wrap with a summary statement regarding how many peaks were detected, etc. -->
<!-- * Might be a good place to suggest next steps (references to chip-qPCR, pathway enrichment, etc.) -->
<!-- * Might also want to make a list of files that will be returned to investigator -->
